src/registrar/models/domain_application.py

```diff
@@ -572,6 +572,19 @@ def domain_is_not_active(self):
             return not self.approved_domain.is_active()
         return True
 
+    def delete_and_clean_up_domain(self, called_from):
+        try:
+            domain_state = self.approved_domain.state
+            # Only reject if it exists on EPP
+            if domain_state != Domain.State.UNKNOWN:
+                self.approved_domain.deletedInEpp()
+                self.approved_domain.save()
+            self.approved_domain.delete()
+            self.approved_domain = None
+        except Exception as err:
+            logger.error(err)
+            logger.error(f"Can't query an approved domain while attempting {called_from}")
```

Comment:
Couldn't you get this from `domain_state`, rather than passing it in?

Reply:
Negatory. This is the transition you're attempting to run. domain_state is the starting point.

Reply:
It is a nice to have though. If you feel strongly about it, we can probably change that logger to something generic and do away with the extra param.

Reply:
Not a big deal! I think having that in the function is useful for the log output (forgot to mark it with "Q"). I looked into this a bit more and the way your doing it makes sense.

Didn't verbalize it well, but I was thinking of the transition tag we have, `@transition(field="state", source=State.UNKNOWN, target=State.DNS_NEEDED)`. I thought maybe the "target" was extractable, but on further research I don't know if thats easy to do without modifying a bazillion functions so definitely not worth pursuing